src/reducers/versions.tsx

```diff
@@ -0,0 +1,218 @@
+import { Reducer } from 'redux';
+import { ActionType, createAction, getType } from 'typesafe-actions';
+
+type VersionId = number;
+
+type LocalizedString = {
+  [lang: string]: string;
+};
+
+type VersionCompatibility = {
+  [appName: string]: {
+    min: string;
+    max: string;
+  };
+};
+
+type VersionLicense = {
+  id: number;
+  isCustom: boolean;
+  name: LocalizedString;
+  text: LocalizedString;
+  url: string;
+};
+
+export type ExternalVersionEntry = {
+  binary: boolean | string;
+  depth: number;
+  directory: boolean;
+  filename: string;
+  mimetype: string;
+  modified: string;
+  path: string;
+  sha256: string;
+  size: number | null;
+};
+
+export type ExternalVersionFile = {
+  content: string;
+  created: string;
+  entries: {
+    [nodeName: string]: ExternalVersionEntry;
+  };
+  hash: string;
+  id: number;
+  is_mozilla_signed_extension: boolean;
+  is_restart_required: boolean;
+  is_webextension: boolean;
+  permissions: string[];
+  platform: string;
+  size: number;
+  status: string;
+  url: string;
+};
+
+export type ExternalVersion = {
+  channel: string;
+  compatibility: VersionCompatibility;
+  edit_url: string;
+  file: ExternalVersionFile;
+  has_been_validated: boolean;
+  id: VersionId;
+  is_strict_compatibility_enabled: boolean;
+  license: VersionLicense;
+  release_notes: LocalizedString | null;
+  reviewed: string;
+  url: string;
+  validation_url: string;
+  validation_url_json: string;
+  version: string;
+};
+
+type VersionFile = {
+  content: string;
+  created: string;
+  id: number;
+  size: number;
+};
+
+export enum VersionEntryType {
+  image,
+  binary,
+  text,
+  directory,
```

Comment:
Great! Can we uppercase them? Those values look like constants to me.

Reply:
They are constants, but I thought the code using them looked better as `VersionEntryType.image` and `VersionEntryType.binary` as opposed to `VersionEntryType.IMAGE` and `VersionEntryType.BINARY`. Can we adopt a convention that when we use `enum` we can use camelCased names?

Reply:
sure